contrib/src/main/java/com/datatorrent/contrib/kinesis/KinesisSinglePortStringInputOperator.java

```
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2013 DataTorrent, Inc. ALL Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datatorrent.contrib.kinesis;
+
+import com.amazonaws.services.kinesis.model.Record;
+
+import java.nio.ByteBuffer;
+
+public class KinesisSinglePortStringInputOperator extends AbstractKinesisSinglePortInputOperator<String>
+{
+
+  /**
+   * Implement abstract method of AbstractKinesisSinglePortInputOperator
+   */
+  @Override
+  public String getTuple(Record record)
+  {
+    String data = "";
+    try {
+      ByteBuffer bb = record.getData();
+      byte[] bytes = new byte[bb.remaining() ];
+      bb.get(bytes);
+      data = new String(bytes);
+    }
+    catch (Exception ex) {
```

Should not swallow exception.